soh/soh/Enhancements/randomizer/randomizer_check_tracker.cpp

```diff
@@ -1019,20 +1540,29 @@ void CheckTrackerWindow::InitElement() {
     Color_Seen_Extra            = CVarGetColor("gCheckTrackerSeenExtraColor",             Color_Seen_Extra_Default);
     Color_Hinted_Main           = CVarGetColor("gCheckTrackerHintedMainColor",            Color_Main_Default);
     Color_Hinted_Extra          = CVarGetColor("gCheckTrackerHintedExtraColor",           Color_Hinted_Extra_Default);
-    Color_Checked_Main          = CVarGetColor("gCheckTrackerCheckedMainColor",           Color_Main_Default);
-    Color_Checked_Extra         = CVarGetColor("gCheckTrackerCheckedExtraColor",          Color_Checked_Extra_Default);
+    Color_Collected_Main        = CVarGetColor("gCheckTrackerCollectedMainColor",         Color_Main_Default);
+    Color_Collected_Extra       = CVarGetColor("gCheckTrackerCollectedExtraColor",        Color_Collected_Extra_Default);
     Color_Scummed_Main          = CVarGetColor("gCheckTrackerScummedMainColor",           Color_Main_Default);
     Color_Scummed_Extra         = CVarGetColor("gCheckTrackerScummedExtraColor",          Color_Scummed_Extra_Default);
     Color_Saved_Main            = CVarGetColor("gCheckTrackerSavedMainColor",             Color_Main_Default);
     Color_Saved_Extra           = CVarGetColor("gCheckTrackerSavedExtraColor",            Color_Saved_Extra_Default);
 
+    SaveManager::Instance->AddInitFunction(InitTrackerData);
+    sectionId = SaveManager::Instance->AddSaveFunction("trackerData", 1, SaveFile, true, -1);
+    SaveManager::Instance->AddLoadFunction("trackerData", 1, LoadFile);
     GameInteractor::Instance->RegisterGameHook<GameInteractor::OnLoadFile>([](uint32_t fileNum) {
         doInitialize = true;
     });
-    GameInteractor::Instance->RegisterGameHook<GameInteractor::OnDeleteFile>([](uint32_t fileNum) {
-        Teardown();
-    });
+    GameInteractor::Instance->RegisterGameHook<GameInteractor::OnDeleteFile>([](uint32_t fileNum) { Teardown(); });
```

Comment:
I know these were here before, but these are the wrong hooks.

For loading you _probably_ want OnLoadGame, but be warned it too has some weird quirks.

For teardown, I'm not even sure we have a proper hook for this, we have OnExitGame but that's only called if they leave after a game over, not on reset.

Maybe after you determine the right place for the `doInitialize = true;`, you can just do a teardown before it eg
```cpp
GameInteractor::Instance->RegisterGameHook<GameInteractor::OnLoadGame>([](uint32_t fileNum) {
  Teardown(); // will need to gracefully handle if it hasn't been initialized before
  doInitialize = true;
```

Reply:
As far as putting `Teardown` with `doInitialize`, `Teardown` is already part of the block called when `doInitialize` is true, so that's not necessary. There was some quirk I had to deal with already in the use of `IsVisibleInCheckTracker` for the mystery item PR which I guess I might as well pull into here anyway (instead of using `doInitialize`, I just converted it to call `Teardown` and `LoadSettings` directly in the hook).

As for the `OnDeleteFile` hook, I *think* you're right. `OnExitGame` would be a better place for that, and I think it has to do with making the check tracker hide the checks in the main menu anyway. I'll look into it again.

Reply:
Alright, looked into it, looks like the `OnLoadFile` hook wasn't even necessary, as Teardown and LoadSettings are already called in `LoadFile` SaveManager hook.